step.rb

```diff
@@ -0,0 +1,276 @@
+require 'pathname'
+
+# -----------------------
+# --- functions
+# -----------------------
+
+def fail_with_message(message)
+  ENV['BITRISE_XAMARIN_TEST_RESULT'] = 'failed'
+
+  puts "\e[31m#{message}\e[0m"
+  exit(1)
+end
+
+def error_with_message(message)
+  puts "\e[31m#{message}\e[0m"
+end
+
+def xcode_major_version
+  out = `xcodebuild -version`
+  begin
+    version = out.split("\n")[0].strip.split(' ')[1].strip.split('.')[0].to_i
+  rescue
+    nil
+  end
+  version
+end
+
+def simulator_udid_and_state(simulator_device, os_version)
+  os_found = false
+  os_regex = "-- #{os_version} --"
+  os_separator_regex = '-- iOS \d.\d --'
+  device_regex = "#{simulator_device}" + '\s*\(([\w|-]*)\)\s*\(([\w]*)\)'
+  out = `xcrun simctl list | grep -i --invert-match 'unavailable'`
+  out.each_line do |line|
+    os_separator_match = line.match(os_separator_regex)
+    os_found = false unless os_separator_match.nil?
+
+    os_match = line.match(os_regex)
+    os_found = true unless os_match.nil?
+
+    next unless os_found
+
+    match = line.match(device_regex)
+    unless match.nil?
+      udid, state = match.captures
+      return udid, state
+    end
+  end
+  nil
+end
+
+def all_simulator_shutteddown
+  all_shotdown = true
+  regex = '\s*\(([\w|-]*)\)\s*\(([\w]*)\)'
+  out = `xcrun simctl list`
+  out.each_line do |line|
+    match = line.match(regex)
+    unless match.nil?
+      _udid, state = match.captures
+      all_shotdown = false if state != 'Shutdown'
+    end
+  end
+  all_shotdown
+end
+
+def export_app(build_path)
+  app_path = Dir[File.join(build_path, '/**/*.app')].first
+  return nil unless app_path
+
+  full_path = Pathname.new(app_path).realpath.to_s
+  return nil unless full_path
+  return nil unless File.exist? full_path
+  full_path
+end
+
+def export_dll(test_build_path)
+  dll_path = Dir[File.join(test_build_path, '/**/*.dll')].first
+  return nil unless dll_path
+
+  full_path = Pathname.new(dll_path).realpath.to_s
+  return nil unless full_path
+  return nil unless File.exist? full_path
+  full_path
+end
+
+def shutdown_simulators(xcode_major_version)
+  all_shotdown = all_simulator_shutteddown
+  return 0 if all_shotdown
+
+  `killall Simulator` if xcode_major_version == 7
+  `killall "iOS Simulator"` if xcode_major_version == 6
+  return 1 unless $?.success?
+
+  loop do
+    sleep 1 # second
+    all_shotdown = all_simulator_shutteddown
+    puts '    => waiting for shutdown ...'
+    break if all_shotdown
+  end
+  0
+end
+
+def boot_simulator(simulator, xcode_major_version)
+  simulator_cmd = ''
```

Comment:
Instead of the empty value, check if it's set at the end and fail if not
`No simulator found`